tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -174,35 +177,46 @@ HRESULT DebuggerMethodRewriter::LoadInstanceIntoStack(FunctionInfo* caller, bool
             //    ldloc.s [localIndex]
-            Logger::Warn("*** DebuggerMethodRewriter::Rewrite() Static methods in a ValueType cannot be instrumented. ");
+            Logger::Warn(
+                "*** DebuggerMethodRewriter::Rewrite() Static methods in a ValueType cannot be instrumented. ");
             return E_FAIL;
         }
         *outLoadArgumentInstr = rewriterWrapper.LoadNull();
     }
     else
     {
-        *outLoadArgumentInstr = rewriterWrapper.LoadArgument(0);
-        if (caller->type.valueType)
+        bool callerTypeIsValueType = caller->type.valueType;
+        mdToken callerTypeToken = callTargetTokens->GetCurrentTypeRef(&caller->type, callerTypeIsValueType);
+        if (callerTypeToken == mdTokenNil)
         {
-            if (caller->type.type_spec != mdTypeSpecNil)
-            {
-                rewriterWrapper.LoadObj(caller->type.type_spec);
-            }
-            else if (!caller->type.isGeneric)
-            {
-                rewriterWrapper.LoadObj(caller->type.id);
-            }
-            else
+            *outLoadArgumentInstr = rewriterWrapper.LoadNull();
+        }
+        else
+        {
+
+            *outLoadArgumentInstr = rewriterWrapper.LoadArgument(0);
+            if (caller->type.valueType)
             {
-                // Generic struct instrumentation is not supported
-                // IMetaDataImport::GetMemberProps and IMetaDataImport::GetMemberRefProps returns
-                // The parent token as mdTypeDef and not as a mdTypeSpec
-                // that's because the method definition is stored in the mdTypeDef
-                // This problem doesn't occur on reference types because in that scenario,
-                // we can always rely on the object's type.
-                // This problem doesn't occur on a class type because we can always relay in the
-                // object type.
-                return E_FAIL;
+                if (caller->type.type_spec != mdTypeSpecNil)
+                {
+                    rewriterWrapper.LoadObj(caller->type.type_spec);
+                }
+                else if (!caller->type.isGeneric)
+                {
+                    rewriterWrapper.LoadObj(caller->type.id);
+                }
+                else
+                {
+                    // Generic struct instrumentation is not supported
+                    // IMetaDataImport::GetMemberProps and IMetaDataImport::GetMemberRefProps returns
+                    // The parent token as mdTypeDef and not as a mdTypeSpec
+                    // that's because the method definition is stored in the mdTypeDef
+                    // The problem is that we don't have the exact Spec of that generic
+                    // We can't emit LoadObj or Box because that would result in an invalid IL.
+                    // This problem doesn't occur on a class type because we can always relay in the
+                    // object type.
```

Comment:
```suggestion
                    // Generic struct instrumentation is currently not supported.
                    // IMetaDataImport::GetMemberProps and IMetaDataImport::GetMemberRefProps return
                    // the parent token as mdTypeDef and not as a mdTypeSpec,
                    // because the method definition is stored in the mdTypeDef.
                    // The problem is that we don't have the exact Spec of that generic instantiation,  and so
                    // we can't emit LoadObj or Box because that would result in producing invalid IL.
                    // This problem doesn't occur on a  reference types because we can always rely on the System.Object type.
```

Reply:
Done in bf6df15958c13137a9c5bbf36c849990bc3d909e.